scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -828,6 +830,44 @@
       (log/error t "In" pool-name "pool, error launching tasks for"
                  (cc/compute-cluster-name compute-cluster) "compute cluster"))))
 
+(defn filter-matches-for-ratelimit
+  "Given a set of matches, determine which compute clusters are beyond the rate limit and filter matches in those compute clusters out."
+  [matches]
+  (let [augmented-matches (->> matches
+                               (group-by match->compute-cluster)
+                               (map
+                                 (fn [[compute-cluster matches-in-compute-cluster]]
+                                   (let [compute-cluster (or compute-cluster )
+                                         compute-cluster-name (if compute-cluster
+                                                                (cc/compute-cluster-name compute-cluster)
+                                                                ; In unit tests we may not have a compute cluster.
+                                                                ; If so, swap this compute cluster in.
+                                                                "fake-compute-cluster-name")
+                                         enforce? (ratelimit/enforce? ratelimit/global-job-launch-rate-limiter)
+                                         token-count (ratelimit/get-token-count! ratelimit/global-job-launch-rate-limiter compute-cluster-name)
+                                         resume-millis (ratelimit/time-until-out-of-debt-millis! ratelimit/global-job-launch-rate-limiter compute-cluster-name)
+                                         skipping-cycle? (and enforce? (neg? token-count))]
+                                     (if skipping-cycle?
+                                       {:skip-rate-limit true
+                                        :why {:matches-skipped (count matches-in-compute-cluster)
+
+                                              :compute-cluster compute-cluster-name
+                                              :tokens-count token-count
+                                              :resume-millis resume-millis}}
+                                       {:skip-rate-limit false
+                                        :matches matches-in-compute-cluster})))))
+        compute-clusters-throttled (->> augmented-matches
+                                        (filter :skip-rate-limit)
+                                        (map :why)
+                                        (reduce conj [] ))
+        compute-clusters-kept (->> augmented-matches
+                                   (remove :skip-rate-limit)
+                                   (map :matches)
+                                   (reduce concat []))]
+    (when-not (empty? compute-clusters-throttled)
+      (log/warn "Skipping a subset of compute clusters because of rate-limit:" compute-clusters-throttled))
```

Comment:
Maybe `Skipping a subset of matches ...`?

Reply:
Changed.